src/main/java/com/minecolonies/coremod/colony/buildings/workerbuildings/BuildingLumberjack.java

```diff
@@ -212,17 +212,19 @@ public AbstractJob createJob(final CitizenData citizen)
     public void readFromNBT(@NotNull final NBTTagCompound compound)
     {
         super.readFromNBT(compound);
-        if (treesToFell.isEmpty())
+        final NBTTagList saplingTagList = compound.getTagList(TAG_SAPLINGS, Constants.NBT.TAG_COMPOUND);
+        for (int i = 0; i < saplingTagList.tagCount(); ++i)
         {
-            final NBTTagList saplingTagList = compound.getTagList(TAG_SAPLINGS, Constants.NBT.TAG_COMPOUND);
-            for (int i = 0; i < saplingTagList.tagCount(); ++i)
+            final NBTTagCompound saplingCompound = saplingTagList.getCompoundTagAt(i);
+            final ItemStack stack = new ItemStack(saplingCompound);
+            final boolean cut = saplingCompound.getBoolean(TAG_CUT);
+            final ItemStorage storage = new ItemStorage(stack); 
+            if (treesToFell.containsKey(storage)) 
             {
-                final NBTTagCompound saplingCompound = saplingTagList.getCompoundTagAt(i);
-                final ItemStack stack = new ItemStack(saplingCompound);
-                final boolean cut = saplingCompound.getBoolean(TAG_CUT);
-                treesToFell.put(new ItemStorage(stack), cut);
-            }
+				treesToFell.put(new ItemStorage(stack), cut);
+			}
```

Comment:
the formatting seems way off here  with the if, the two brackets should be on the same line and the code way to the left.